client/state/data-layer/wpcom-http/pipeline/retry-on-failure/README.md

```diff
@@ -0,0 +1,111 @@
+This means that _nothing needs to be done explicitly in order to take advantage of this retry mechanism_.
+Custom policies can specify that we don't want to attempt a retry at all; and we can change the initial delay and max retry count.
+These are added with the `retryPolicy` override in the HTTP request description.
+
+```js
+// stop retry attempts
+import { noRetry } from 'state/data-layer/wpcom-http/pipeline/retry-on-failure/policies';
+
+dispatch( http( {
+	path: '/sites',
+	method: 'GET',
+	onSuccess,
+	onFailure,
+	retryPolicy: noRetry(),
+}, action );
+
+// moar attempts for a notoriously slow and buggy server
+import { exponentialBackoff } from 'state/data-layer/wpcom-http/pipeline/retry-on-failure/policies';
+
+dispatch( http( {
+	path: '/sites',
+	method: 'GET',
+	onSuccess,
+	onFailure,
+	retryPolicy: exponentialBackoff( { delay: 4000, maxAttempts: 5 } ),
+}, action );
+```
+
+## Delay calculation
+
+The delay algorithm is designed to mitigate congestion which can occur as a side-effect of simple retry delays.
+For example, if a specific API endpoint were to unexpectedly fail and cascade to ten thousand clients, we would want to make sure that those ten thousand clients don't all attempt their retry at or around the same time.
+By introducing pseudo-randomness and jitter into the calculated delay times we can spread out the failure resolution and prevent further failure cascades.
+
+The following histograms represent trial calculations for retry attempts numbered by graph, each containing five thousand delay computations.
+The shapes of the histograms illustrate what kinds of delay values are typical, normalized to the base delay.
+
+<img src="https://cldup.com/Rf-Emw5MJ2.png" />
+<img src="https://cldup.com/1okGMI4UDk.png" />
+<img src="https://cldup.com/fd4wocHuWF.png" />
+<img src="https://cldup.com/3PeG4ELGck.png" />
+<img src="https://cldup.com/B3Bl8rKnO2.png" />
+
+Surprisingly we can see from the graphs that it's possible for later retry attempts to have shorter delays than earlier retry attempts.
+However, if we examine what happens over the average total delay time when a request resolves after a given number of retries we can see a different picture.
+These next histograms show the total delay time for all requests given the number of retries, each with five thousand trial calculations.
+
+<img src="https://cldup.com/xvXKvAQ2tr.png" />
+<img src="https://cldup.com/yvt0G3KnEg.png" />
+<img src="https://cldup.com/0TcRLxeMf4.png" />
+<img src="https://cldup.com/z52VCOosOM.png" />
+
+The entire expected wait time corresponds to an exponential shape.
+That is to say, we expect the total wait time to be exponential with the number of retry attempts it requires before resolution.
+The following table summaries then total average wait times for when a request resolves after the given number of retries, normalized to the base delay.
+
+| # Retries | Expected Wait Time |
+| --- | --- |
+| 1 | 1.627 |
```

Comment:
does this take into account slow internet connections? If the delays are calculated from the time the response is received then actual timings could be slower